spec/forms/provider_interface/provider_user_providers_form_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe ProviderInterface::ProviderUserProvidersForm do
+  describe '#save' do
+    it 'prevents adding people to organisations they do not have access to' do
+      provider_user = create(:provider_user)
+
+      provider = create(:provider)
+      provider_i_dont_have_access_to = create(:provider)
```

Comment:
What happens if the current user is associated with a provider but does not have `manage_users` permission for that provider?

Reply:
Ok, turns out we only show the intersection of providers common to both current_provider_user and the user in question, __if__ the current user has `manage_users`. This is because `possible_providers` has a built-in `.manage_users` scope. It's not very clear, perhaps could do with a refactor.

Reply:
I'm not sure I understand, where would you do a refactor?

Reply:
Sorry, not related to this PR, I think we should rename/move `possible_providers`.

Reply:
Done!